.github/workflows/ci.yml

```diff
@@ -263,71 +272,82 @@ jobs:
   docker:
-    name: Docker
-    # Don't run nightly builds for forks.
-    if: github.repository == 'allenai/allennlp' || github.event_name != 'schedule'
-    needs: [build_package]  # needs the package artifact created from 'build_package' job.
-    runs-on: ubuntu-latest
+    name: Docker (CUDA ${{ matrix.cuda }})
+    if: github.repository == 'allenai/allennlp'
+    runs-on: [self-hosted, GPU]
+    strategy:
+      matrix:
+        cuda: ['10.2', '11.0']
 
     steps:
     - uses: actions/checkout@v2
 
-    - name: Set Docker image name
+    - name: Set torch version
+      env:
+        CUDA: ${{ matrix.cuda }}
       run: |
-        if [[ $GITHUB_EVENT_NAME == 'release' ]]; then
-            echo "DOCKER_IMAGE_NAME=allennlp/allennlp:${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV;
+        # Check the install instructions on https://pytorch.org/ to keep these up-to-date.
+        if [[ $CUDA == '10.2' ]]; then
+            echo "DOCKER_TORCH_VERSION='torch==1.7.0'" >> $GITHUB_ENV;
+        elif [[ $CUDA == '11.0' ]]; then
+            echo "DOCKER_TORCH_VERSION='torch==1.7.0+cu110 -f https://download.pytorch.org/whl/torch_stable.html'" >> $GITHUB_ENV;
         else
-            echo "DOCKER_IMAGE_NAME=allennlp/commit:$GITHUB_SHA" >> $GITHUB_ENV;
+            echo "Unhandled CUDA version $CUDA";
+            exit 1;
         fi
 
-    - name: Download core package
-      uses: actions/download-artifact@v1
-      with:
-        name: core-package
-        path: dist
+    - name: Set image name
+      env:
+        CUDA: ${{ matrix.cuda }}
+      run: |
+        if [[ $GITHUB_EVENT_NAME == 'release' ]]; then
+            echo "DOCKER_IMAGE_NAME=allennlp/allennlp:${GITHUB_REF#refs/tags/}-cuda${CUDA}" >> $GITHUB_ENV;
+        else
+            echo "DOCKER_IMAGE_NAME=allennlp/commit:${GITHUB_SHA}-cuda${CUDA}" >> $GITHUB_ENV;
+        fi
 
     - name: Build image
       run: |
-        make docker-image DOCKER_IMAGE_NAME=$DOCKER_IMAGE_NAME
+        make docker-image DOCKER_IMAGE_NAME="$DOCKER_IMAGE_NAME" DOCKER_TORCH_VERSION="$DOCKER_TORCH_VERSION"
 
     - name: Test image
       run: |
         make docker-run DOCKER_IMAGE_NAME=$DOCKER_IMAGE_NAME ARGS='test-install'
 
-    - name: Upload master image
-      # Only run this for pushes to master on the main repo, not forks.
-      if: github.repository == 'allenai/allennlp' && github.event_name == 'push'
+    - name: Upload commit image
+      # Only upload the 10.2 image.
+      if: github.event_name == 'push' && matrix.cuda == '10.2'
       run: |
         docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
         docker push $DOCKER_IMAGE_NAME
 
     - name: Upload release image
-      # Only run this for releases on the main repo, not forks.
-      if: github.repository == 'allenai/allennlp' && github.event_name == 'release'
+      # Only run this for releases.
+      if: github.event_name == 'release'
```

Comment:
Why did all of these `if` statements change?

Reply:
We checked if `github.repository == 'allenai/allennlp'` above, so it was redundant to have that again.